### PR TITLE
feat: add Debian packaging

### PR DESCRIPTION
**Pull request by enneamer**
I have tried to build it with git-buildpackage on Debian Bookworm, and it works. Maybe it needs some linting, but I think we can save it for later.

### CODE REVIEW

File: debian/copyright [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Source: <url://example.com>
+Upstream-Name: dracut-sshd
+Upstream-Contact: <preferred name and address to reach the upstream project>
+
+Files:
+ *
+Copyright:
+ <years> <put author's name and email here>
+ <years> <likewise for another author>
+License: GPL-3.0+
```
Comment (gsauthof) [inline debian/copyright:11]: This specifies what license is used for the project?

File: debian/watch [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# Compulsory line, this is a version 4 file.
+version=4
+
+# PGP signature mangle, so foo.tar.gz has foo.tar.gz.sig.
+#opts="pgpsigurlmangle=s%$%.sig%"
+
+# HTTP site (basic).
+#http://example.com/downloads.html \
+#    files/dracut-sshd-([\d\.]+)\.tar\.gz
```
Comment (gsauthof) [inline debian/watch:14]: Do we need all the commented out stuff?

File: debian/control [hunk 3]
```diff
@@ -0,0 +1,20 @@
+Source: dracut-sshd
+Section: contrib/utils
+Priority: optional
+Maintainer: Georg Sauthoff <mail@georg.so>
```
Comment (gsauthof) [inline debian/control:4]: Is this the maintainer of the software itself or the debian package?

**Review by gsauthof — COMMENTED** [review]
Do you plan to integrate such a package into the debian distribution?

What are the steps to build such a package?

**Comment by enneamer** [comment]
Thank you for your review, @gsauthof .

> Do you plan to integrate such a package into the debian distribution?

Sorry that I did not make my intention clear. In this pull request, I merely add necessary code for Debian packaging. This can help people who wish to build a Debian/Ubuntu package to deploy on their own machines, especially when they have many machines to manage and wish to share a system package for easier deployment. In addition, in case this project is to adopt GitHub Actions to offer pre-built packages in the future, this PR might be helpful. (I am aware that there are solutions to packages for many distros. I am not sure how well they would work for system packages like this, which might need distro-specific modifications.)

> What are the steps to build such a package?

I should have included information like this. Do you mind I put it in README.md in the PR?

File: debian/copyright [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+Files:
+ debian/*
+Copyright:
+ 2018-2023 Georg Sauthoff <mail@gms.tf>
+License: GPL-3.0+
```
[the remark this answers is not in the document]
Comment (enneamer) [reply debian/copyright:10]: I have merged this with all-file copyright section.

File: debian/copyright [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+License: GPL-3.0+
```
[the remark this answers is not in the document]
Comment (enneamer) [reply debian/copyright:12]: I believe the second `License` line introduces the license content. The examples in Debian packaging manual [1] seem to be in the same style.

[1]: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/

File: debian/copyright (hunk 1 above)
Comment (enneamer) [reply debian/copyright]: My bad. I have fixed this section.

File: debian/watch (hunk 2 above)
Comment (enneamer) [reply debian/watch]: I have removed. Thanks.

File: debian/control (hunk 3 above)
Comment (enneamer) [reply debian/control]: It indeed should be the package maintainer's information. I have changed it to my info. Please confirm whether this works for you.

**Comment by enneamer** [comment]
I have added a packaging guide in README.md.

File: debian/control (hunk 3 above)
Comment (gsauthof) [reply debian/control]: Yes, I don't have any plans for maintaining a debian package.

File: debian/copyright (hunk 5 above)
[the remark this answers is not in the document]
Comment (gsauthof) [reply debian/copyright:18]: Hm, feels like overkill. I mean I don't even include a copy of the license text for dracut-sshd proper.

I think that is sufficient to use a standardized license tag. And possibly link to the canonical license source.

Debian might have some strict policy when it comes to adding such a package to their repositories.
However, when that happens that can be added then in debian.

IIRC, when it comes to packaging for debian, debian encourages packagers to keep source package files in a repository separate from the upstream source, anyways.

File: README.md [hunk 6]
```diff
@@ -534,6 +534,24 @@ Related Fedora ticket: [Bug 524727 - Dracut + encrypted root + networking (2009)
...
+   with `docker run -i -t -w /root debian:stable-slim`
+2. Install `git-buildpackage`: `apt-get update && apt-get install git-buildpackage`
+3. Clone the repository: `gbp clone https://github.com/gsauthof/dracut-sshd.git`
+4. Enter the folder: `cd dracut-sshd`
+5. Build the package: `gbp buildpackage --git-upstream-tree=branch --no-sign`
+6. The debian packages and relavant files should be in the parent folder.
+
+For more information regarding proper Debian packaging, please refer to
+[Debian wiki page for packaging with Git][debian-package-git] and
+[Git-buildpackage official website][git-buildpackage].
```
Comment (gsauthof) [inline README.md:554]: I prefer to  move this to a small README.md in that debian subdirectory.

It feels like this debian packaging intro is only remotely related to the topic of the main README which is already long enough.

**Comment by gsauthof** [comment]
Say I'll merge your pull-request - would you be available to help out with issues/pull-request reported on that debian packaging (in case there are any)?

Ideally, having the debian packaging stuff in this repository would be just a temporary solution until a debian packager adds a dracut-sshd package to the official Debian repository.
Then the debian packaging files could be maintained outside of this repository.

FWIW, I packaged some software for debian more than a decade ago but repressed most memories.
Especially the debian way of splitting the packaging specification over a multitude of files annoyed me very much.

For Fedora and other RPM distributions (where dracut-sshd isn't  an official package yet) I'm using the COPR service.
Is there something similar for debian?

File: README.md (hunk 6 above)
Comment (enneamer) [reply README.md]: Moved to debian/README.md

**Comment by enneamer** [comment]
I'll be glad to deal with issues and PRs related to Debian packaging. I have added relavant information in `debian/README.md`.

And I agree that it would be better to have a Debian maintainer to take on the work. Yet I haven't seen a packaging request on bugs.debian.org , so I guess it won't happen very soon.

As for third-party package hosting, Debian itself does not have an official platform similar to COPR to my best knowledge. Ubuntu has PPAs, and many contributors publish their Debian packages there as well. For non-official ones, packagecloud.io provides unified hosting across many distros, including DEB and RPM.

**Comment by gsauthof** [comment]
Ok, I came to the conclusion that this repository isn't the best place for debian packaging files.

If not already present I would even put the RPM packaging into a separate repository, today.

So the alternatives are:

1. try to get dracut-sshd into the official Debian package repository
2. create a separate github repository for debian packaging

Either way, I'm happy to link any such effort from the README, if something materializes.

**Comment by slonkazoid** [comment]
> For Fedora and other RPM distributions (where dracut-sshd isn't an official package yet) I'm using the COPR service.
Is there something similar for debian?

OpenSuse's OBS provides packaging for Debian as well

**Pull request closed without merging**